trie/smt.py

```diff
@@ -0,0 +1,355 @@
+from typing import (
+    Dict,
+    Sequence,
+    Tuple,
+)
+
+import copy
+
+from eth_utils import (
+    keccak,
+    to_int,
+)
+
+from eth_typing import (
+    Hash32,
+)
+
+from trie.constants import (
+    BLANK_NODE,
+)
+
+from trie.exceptions import (
+    ValidationError,
+)
+
+from trie.validation import (
+    validate_is_bytes,
+    validate_length,
+)
+
+
+def calc_root(key: bytes, value: bytes, branch: Sequence[Hash32]) -> Hash32:
+    """
+    Obtain the merkle root of a given key/value/branch set.
+    Can be used to validate a merkle proof or compute it's value from data.
+
+    :param key: the keypath to decide the ordering of the siblings in the branch
+    :param value: the value (or leaf) that starts the merkle proof computation
+    :param branch: the sequence of siblings used to recursively perform the computation
+
+    :return: the root hash of the merkle proof computation
+
+    .. doctest::
```

Comment:
Nope lol.

I could turn this into an actual test case

Reply:
So, these examples aren't too substantially different from the test cases to warrant creating more. I could remove it from the docstring to prevent an out-of-date example from weaseling it's way in.